packages/react-ui/src/widgets/legend/LegendWrapper.js

```diff
@@ -150,3 +192,77 @@ function Header({
     </Grid>
   );
 }
+
+const useOpacityControlStyles = makeStyles(({ spacing }) => ({
+  content: {
+    height: 'auto',
+    flex: 1
+  },
+  slider: {
+    marginTop: spacing(1)
+  },
+  input: {
+    width: spacing(8),
+    '& input': {
+      '&[type=number]': {
+        appearance: 'textfield'
+      },
+      '&::-webkit-outer-spin-button': {
+        appearance: 'none',
+        margin: 0
+      },
+      '&::-webkit-inner-spin-button': {
+        appearance: 'none',
+        margin: 0
+      }
+    },
+    '& .MuiInputAdornment-positionEnd': {
+      margin: 0
+    }
+  }
+}));
+
+function OpacityControl({ opacity, onChangeOpacity }) {
```

Comment:
I know this was already like this, all functions inside same file, but I think it will probably make sense to create a folder, and a separated file per component, what do you think?

Reply:
I agree. I'll modify it.